### PR TITLE
Move `--show-viewer` option to testsupport, add test

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
This fixes an issue using our `make_test_viewer` externally raised by @joshmoore in https://github.com/napari/napari/pull/2131#issuecomment-773215013
(basically, the `--show-viewer` argument didn't make it into the _testsupport.py file)

It also adds a test using [pytester](https://docs.pytest.org/en/stable/writing_plugins.html#testing-plugins) to make sure that our plugin works as expected

Would be nice to get this out ASAP... cc @sofroniewn 

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

### CODE REVIEW

**Comment by sofroniewn** [comment]
@tlambert03 change looks good to me, but looks like we're getting some test fails in

- PR Test / ubuntu-18.04 3.8 headless-py38-linux (pull_request) Failing after 2m — ubuntu-18.04 3.8 headless-py38-linux

- PR Test / ubuntu-latest 3.8 pip install (pull_request) Failing after 4m — ubuntu-latest 3.8 pip install

> Would be nice to get this out ASAP... cc @sofroniewn

Should we try and get 0.4.5 released too asap, or just the merge?

**Comment by tlambert03** [comment]
> but looks like we're getting some test fails in

will fix.

> Should we try and get 0.4.5 released too asap, or just the merge?

well, I don't know.  There _is_ a workaround, before 0.4.5 comes out, I think people can still use the fixture by creating a conftest.py in their repo with:

```python
#conftest.py
def pytest_addoption(parser):
    parser.addoption("--show-viewer", action="store_true", default=False)
```

... but otherwise `make_napari_viewer` won't work outside of our repo

**Comment by tlambert03** [comment]
@sofroniewn I need some input here.  The best way to test our pytest plugin is to use the [`pytester` plugin](https://docs.pytest.org/en/stable/writing_plugins.html#testing-plugins) from pytest, which I'm doing here.  You "use" that either by adding `-p pytester` to the command line args, or by putting `pytest_plugins = ["pytester"]` in a _top level_ conftest.py. We do not use a top-level `conftest.py` (because we wanted to easily test our sdist), so I added that `-p pytester` flag to our pytest addopts in pyproject.toml.  So on CI and locally, you can still just run "pytest".

I'm hesitant to make it such that we have to remember to use `pytest --pyargs -p pytester napari` when we test the pip installed package though.  So, here are the options:

1. just require the user to remember use `-p pytester` when they test our sdist with `--pyargs`
2. exclude `test_testsupport.py` from the sdist in our MANIFEST.in so it doesn't make it into the distribution (and therefore pytester isn't required)
3. move `test_testsupport.py` to the root level (or a `tests/` folder at the root level that doesn't get put into the distribution.

**Comment by sofroniewn** [comment]
> So, here are the options:
>
> 1. just require the user to remember use -p pytester when they test our sdist with --pyargs
> 2. exclude test_testsupport.py from the sdist in our MANIFEST.in so it doesn't make it into the distribution (and therefore pytester isn't required)
> 3. move test_testsupport.py to the root level (or a tests/ folder at the root level that doesn't get put into the distribution.

I lean towards option 2.

I think option 1. seems like a lot to require, and I don't love the idea of more top level files for option 3.

While option 2. might feel like a bit of a special case I think that's fine, this is a special case :-)

Hope this perspective helps!!

**Comment by tlambert03** [comment]
well... I tried option 2, but couldn't get it to work, because, amazingly, it's still hard to [exclude a single file from a wheel](https://github.com/pypa/setuptools/issues/511).  I'm putting it in root for now.  If anyone is motivated to move it, we can address it later.

**Comment by tlambert03** [comment]
ugh... that strategy is failing to.  going to pick this up later

File: test_pytest_plugin.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+This module tests our "pytest plugin" made available in
+``napari.utils._testsupport``.  We leave this file in root so it is not
+included in the sdist OR the wheel.
+"""
+
+import pytest
+
+
+@pytest.mark.filterwarnings("ignore:`type` argument to addoption()::")
+@pytest.mark.filterwarnings("ignore:The TerminalReporter.writer::")
```
Comment (Carreau) [inline test_pytest_plugin.py]: Ha, I didn't know about this fixture; nice.

**Review by Carreau — APPROVED** [review]

**Comment by tlambert03** [comment]
found a better solution (putting `pytest_plugins = "pytester"` in the module itself), this is ready now @sofroniewn

File: napari/_tests/test_pytest_plugin.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+"""
+This module tests our "pytest plugin" made available in
```
Comment (sofroniewn) [inline napari/_tests/test_pytest_plugin.py:2]: Great comment!

**Review by sofroniewn — APPROVED** [review]
LGTM! Nice job

**Pull request merged**